cli/pkg/config/cli_config.go

```diff
@@ -43,11 +61,28 @@ func (c *CLIConfigManager) LoadCLIConfig() (CLIConfig, error) {
 		return cliConfig, nil
 	}
 
-	data, err := fileutils.ReadFile(c.CLIConfigPath)
-	if err != nil {
+	// Allow setting file path after CLIConfigManager is initialized
+	c.viper.SetConfigFile(c.CLIConfigPath)
+	if err := c.viper.ReadInConfig(); err != nil {
 		return cliConfig, fmt.Errorf("error when reading config file: %w", err)
 	}
-	if err := json.Unmarshal(data, &cliConfig); err != nil {
+
+	if err := c.viper.Unmarshal(&cliConfig, func(dConfig *mapstructure.DecoderConfig) {
+		dConfig.DecodeHook = mapstructure.StringToTimeHookFunc(time.RFC3339)
```

Comment:
[`StringToTimeHookFunc`](https://pkg.go.dev/github.com/mitchellh/mapstructure#StringToTimeHookFunc) decode hook was added because `viper.Unmarshal` does not know how to unmarshal json time string into `*time.Time` (it throws an error if we don't add the decode hook)